src/SmartComponents/Systems/SystemsListAssets.js

```diff
@@ -164,8 +166,48 @@ const isPatchSetRemovalDisabled = (row) => {
     return !baselineName || (typeof baselineName === 'string' && baselineName === '');
 };
 
+export const useActivateRemediationModal = (setRemediationModalCmp, setRemediationOpen) => {
+    const { fetchBatched } = useFetchBatched();
+
+    return useCallback(async (rowData) => {
+        const filter = {
+            id: rowData.id,
+            'filter[status]': 'in:Installable'
+        };
+
+        const totalCount = await fetchApplicableSystemAdvisoriesApi({ ...filter, limit: 1 }).then(
+            (response) => response?.meta?.total_items || 0
+        );
+
+        fetchBatched(
+            (__, pagination) => fetchApplicableSystemAdvisoriesApi({ ...filter, ...pagination }),
+            totalCount,
+            filter
+        ).then(response => {
+            const advisories = response.flatMap(({ data }) => data);
+            const remediationIssues = remediationProvider(
+                advisories.map(item => item.id),
+                rowData.id,
+                remediationIdentifiers.advisory
+            );
+
+            setRemediationModalCmp(() =>
```

Comment:
In the past we have seen issues with putting components into redux or a state.

Maybe we can look into rethinking this and avoid putting the component in here.